File: frontend/src/features/jobs/types/job.ts
export type JobStatus =
  | "queued"
  | "running"
  | "succeeded"
  | "failed"
  | "cancelled";

export interface Job {
  id: number;
  pipelineId: number;
  status: JobStatus;
  startedAt: number | null;
  finishedAt: number | null;
  errorMessage: string;
  createdAt: number;
  updatedAt: number;
}

File: backend/internal/model/job.go
package model

type JobStatus string

const (
	JobStatusQueued    JobStatus = "queued"
	JobStatusRunning   JobStatus = "running"
	JobStatusSucceeded JobStatus = "succeeded"
	JobStatusFailed    JobStatus = "failed"
	JobStatusCancelled JobStatus = "cancelled"
)

type Job struct {
	BaseModel

	PipelineID uint `gorm:"not null;index" json:"pipelineId"`

	Pipeline Pipeline `gorm:"constraint:OnUpdate:CASCADE,OnDelete:CASCADE;" json:"-"`

	Status JobStatus `gorm:"type:varchar(20);not null;default:'queued';index" json:"status"`

	StartedAt *int64 `json:"startedAt"`

	FinishedAt *int64 `json:"finishedAt"`

	ErrorMessage string `gorm:"type:text" json:"errorMessage"`
}
